# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [3]:
import pandas as pd
import numpy as np
from scipy import stats

In [4]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [5]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [6]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [9]:
# number of callbacks for white-sounding names
sum(data[data.race=='w'].call)

235.0

In [65]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4870 entries, 0 to 4869
Data columns (total 65 columns):
id                    4870 non-null object
ad                    4870 non-null object
education             4870 non-null int8
ofjobs                4870 non-null int8
yearsexp              4870 non-null int8
honors                4870 non-null int8
volunteer             4870 non-null int8
military              4870 non-null int8
empholes              4870 non-null int8
occupspecific         4870 non-null int16
occupbroad            4870 non-null int8
workinschool          4870 non-null int8
email                 4870 non-null int8
computerskills        4870 non-null int8
specialskills         4870 non-null int8
firstname             4870 non-null object
sex                   4870 non-null object
race                  4870 non-null object
h                     4870 non-null float32
l                     4870 non-null float32
call                  4870 non-null float32
city        

In [7]:
data.describe()

education       ofjobs     yearsexp       honors    volunteer  \
count  4870.000000  4870.000000  4870.000000  4870.000000  4870.000000   
mean      3.618480     3.661396     7.842916     0.052772     0.411499   
std       0.714997     1.219126     5.044612     0.223601     0.492156   
min       0.000000     1.000000     1.000000     0.000000     0.000000   
25%       3.000000     3.000000     5.000000     0.000000     0.000000   
50%       4.000000     4.000000     6.000000     0.000000     0.000000   
75%       4.000000     4.000000     9.000000     0.000000     1.000000   
max       4.000000     7.000000    44.000000     1.000000     1.000000   

          military     empholes  occupspecific   occupbroad  workinschool  \
count  4870.000000  4870.000000    4870.000000  4870.000000   4870.000000   
mean      0.097125     0.448049     215.637782     3.481520      0.559548   
std       0.296159     0.497345     148.127551     2.038036      0.496492   
min       0.000000     0.000000       7.000000     1.000000      0.000000   
25%       0.000000     0.000000      27.000000     1.000000      0.000000   
50%       0.000000     0.000000     267.000000     4.000000      1.000000   
75%       0.000000     1.000000     313.000000     6.000000      1.000000   
max       1.000000     1.000000     903.000000     6.000000      1.000000   

          ...           educreq      compreq       orgreq        manuf  \
count     ...       4870.000000  4870.000000  4870.000000  4870.000000   
mean      ...          0.106776     0.437166     0.072690     0.082957   
std       ...          0.308866     0.496083     0.259649     0.275854   
min       ...          0.000000     0.000000     0.000000     0.000000   
25%       ...          0.000000     0.000000     0.000000     0.000000   
50%       ...          0.000000     0.000000     0.000000     0.000000   
75%       ...          0.000000     1.000000     0.000000     0.000000   
max       ...          1.000000     1.000000     1.000000     1.000000   

          transcom     bankreal        trade   busservice   othservice  \
count  4870.000000  4870.000000  4870.000000  4870.000000  4870.000000   
mean      0.030390     0.085010     0.213963     0.267762     0.154825   
std       0.171677     0.278932     0.410141     0.442847     0.361773   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     1.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

           missind  
count  4870.000000  
mean      0.165092  
std       0.371308  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max       1.000000  

[8 rows x 55 columns]

# 1) What test is appropriate for this problem? Does CLT apply?
The goal is to test whether race has a significant impact on the rate of callbacks for resumes. We can test the equality of two proportions of callback rates for both races. Since there are 4870 resumes, the central limit theorem applies and states that the difference between the two proportions is approximately a normal distribution. 


# 2) What are the null and alternate hypotheses?
Null Hypothesis Ho: The proportion of callbacks for black-sounding names is equal to the proportion of callbacks for white-sounding names

Alternate Hypothesis Ha: The proportion of callbacks for black-sounding names is not equal to the proportion of callbacks for white-sounding names


In [26]:
nblack=len(data[data.race=='b'].call)
nwhite=len(data[data.race=='w'].call)

In [27]:
sumblack=sum(data[data.race=='b'].call)
sumwhite=sum(data[data.race=='w'].call)

In [31]:
# Compute the callback rates for both races
pwhite= sumwhite/nwhite
pblack= sumblack/nblack
print('Callbackrates for blacks are:',pblack)
print('Callbackrates for whites are:',pwhite)

Callbackrates for blacks are: 0.064476386037
Callbackrates for whites are: 0.0965092402464


In [32]:
print('Callback rates for blacks & whites are: {:1.3f}, {:1.3f}'.format(pblack,pwhite))

Callback rates for blacks & whites are: 0.064, 0.097


# 3)Compute margin of error, confidence interval, and p-value.

In [13]:
SE_black = (pblack * (1 - pblack)) / nblack
SE_white = (pwhite * (1 - pwhite)) / nwhite


In [14]:
SE = np.sqrt(SE_black + SE_white)
SE

0.0077833705866767544

# Margin of error
For 95% confidence interval, the critical z value is 1.96. We can calculate the Margin of error by multiplying this critical value with the standard error(SE).


In [38]:
ME1= 1.96*SE
ME1

0.015255406349886438

For 99% confidence interval, the critical z value is 2.576. We can calculate the Margin of error by multiplying this critical value with the standard error(SE).

In [39]:
ME2= 2.576*SE
ME2

0.020049962631279322

# Calculating the 95% confidence interval we get,

In [44]:
diff=pwhite-pblack
CI=[diff-ME1, diff+ME1]
print('95% confidence interval:',CI)

95% confidence interval: [0.016777447859559147, 0.047288260559332024]


# Calculating the 99% confidence interval we get,

In [45]:
diff=pwhite-pblack
CI=[diff-ME2, diff+ME2]
print('99% confidence interval:',CI)

99% confidence interval: [0.011982891578166264, 0.05208281684072491]


# p-value

In [49]:
z= (diff-0)/SE
z

4.1155504357300003

In [51]:
pval = 2*(1 - stats.norm.cdf(np.abs(z)))
pval

3.862565207524149e-05

# 4) Write a story describing the statistical significance in the context or the original problem.

Resumes with black-sounding names receive a callback rate of 6.4%, while white names receive a callback rate of 9.7%. This difference of 3.3 percentage points is statistically significant, 
Since the p value is almost 0, which is less than 0.01, We reject null hypothesis in favor of alternative hypothesis. Therefore, racial discrmination in the U.S. labor market still appears to be a continual challenge. 


# 5) Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

The analysis shows that racial soundingness of names has a significant effect on callbacks for interviews, but there are also other factors, such as education, years of experience and computer skills, that can affect the callback success. A logistic regression is needed to examine which factors are significant predictors and their effect sizes. 